apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -97,6 +97,7 @@ export interface IRushConfigurationJson {
   eventHooks?: IEventHooksJson;
   hotfixChangeEnabled?: boolean;
   pnpmOptions?: IPnpmOptionsJson;
+  requireVersionChecks?: boolean;
```

Comment:
>requireVersionChecks [](start = 2, length = 20)

I think we should distinguish two concepts:

"rush check" is a command that prints a report about various possible inconsistencies in the package.json files.  Some people may consider some inconsistencies to be an error.

"enforce consistent versions" is an optional policy.  When enabled, Rush reports errors to block changes that would violate the policy.  There will various options for exceptions to the policy, e.g. allowedAlternativeVersions.

Probably we can find a better name for the policy.  I'm not sure we'd want it enabled by default. There's a valid scenario where someone has 50 projects and simply wants to build them in bulk.  That's different from our common scenario where we develop 50 projects as an integrated set of work.